cpp/src/arrow/engine/substrait/relation_internal.cc

```diff
@@ -48,10 +48,9 @@ struct EmitInfo {
   std::shared_ptr<Schema> schema;
 };
 
-template <typename RelMessage>
-Result<EmitInfo> GetEmitInfo(const RelMessage& rel,
```

Comment:
Can you explain the change from RelMessage -> RelCommon?

Reply:
This change simplifies the handling of extension messages, of which there are 3 cases: `ExtenionLeaf`, `ExtensionSingle`, and `ExtensionMulti`. These are 3 different types which in the pre-RP code required 3 code paths to handle. Since this handling only cares about the `RelCommon` part of these messages, the PR code refactored the handling to a common code path and localized the difference into `GetExtensionRelCommon`.